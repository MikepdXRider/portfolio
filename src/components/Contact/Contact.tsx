import React from 'react';

interface ContactProps {
  imgSrc: string;
  imgAlt: string;
  linkHref: string;
  linkAriaLabel: string;
}

export default function Contact({
  imgSrc,
  imgAlt,
  linkHref,
  linkAriaLabel,
}: ContactProps) {
  return (
    <article>
      <img src={imgSrc} alt={imgAlt} />
      <a href={linkHref} aria-label={linkAriaLabel}>
        {linkHref}
      </a>
    </article>
  );
}
