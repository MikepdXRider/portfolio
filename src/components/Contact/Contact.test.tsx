import { render } from '@testing-library/react';
import Contact from './Contact';

describe('contact component', () => {
  const mockContactProps = {
    imgSrc: 'https://placekitten.com',
    imgAlt: 'mock alt',
    linkHref: 'https://placecage.com',
    linkAriaLabel: 'mock link aria-label ',
  };

  it('matches the existing snapshot', () => {
    const { container } = render(<Contact {...mockContactProps} />);

    expect(container).toMatchSnapshot();
  });
});
