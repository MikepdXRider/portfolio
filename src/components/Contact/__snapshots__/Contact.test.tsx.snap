// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`contact component matches the existing snapshot 1`] = `
<div>
  <article>
    <img
      alt="mock alt"
      src="https://placekitten.com"
    />
    <a
      aria-label="mock link aria-label "
      href="https://placecage.com"
    >
      https://placecage.com
    </a>
  </article>
</div>
`;
